flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/exceptionhistory/ExceptionHistoryEntryExtractor.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.exceptionhistory;
+
+import org.apache.flink.runtime.executiongraph.ExecutionJobVertex;
+import org.apache.flink.runtime.executiongraph.ExecutionVertex;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.scheduler.strategy.ExecutionVertexID;
+import org.apache.flink.util.Preconditions;
+
+import java.util.Map;
+
+/**
+ * {@code ExceptionHistoryEntryExtractor} extracts all the necessary information from given
+ * executions to create corresponding {@link RootExceptionHistoryEntry RootExceptionHistoryEntries}.
+ */
+public class ExceptionHistoryEntryExtractor {
+
+    public RootExceptionHistoryEntry extractLocalFailure(
+            Map<JobVertexID, ExecutionJobVertex> executionJobVertices,
+            ExecutionVertexID executionVertexId,
+            Iterable<ExecutionVertexID> otherAffectedVertices) {
+        final ExceptionHistoryEntry entry =
+                extractExceptionHistoryEntry(executionJobVertices, executionVertexId);
+
+        Preconditions.checkArgument(
+                entry != null,
+                "The passed ExecutionVertexId {} does not correspond to a failed execution.",
```

Comment:
this can be misleading; e.g., if executionJobVertices does not contain the vertex.

Reply:
This is fixed by adding explicit failures 👍